contracts/v2/core/SharedDepositMinterV2.sol

```diff
@@ -280,7 +280,9 @@ contract SharedDepositMinterV2 is AccessControl, Pausable, ReentrancyGuard, ETH2
         Address.sendValue(recv, assets);
     }
 
-    receive() external payable {} // solhint-disable-line
+    receive() external payable {
+        _depositAccounting();
```

Comment:
should not be running deposit accounting on receive. 
tokens sent directly to the contract without a fn call should be treated as donations and not mint any new derivative w/sgeth tokens

Reply:
![image](https://github.com/chimera-defi/SharedDeposit/assets/101313445/1cff2ce3-52cc-41e5-b5fb-d4ba8f2101e7)
This should be increase curValidatorShares, otherwise it make panic error though contract has enough balance.

Reply:
hmm ill take a look.  shouldnt do that. 
so 100 E flows in.  + 100E
96 E flows to validators  - 96
user requests 32 E back -32 
validator is exited, funds are directed to withdrawalQueue +32

so curValidatorShares should never go above ~100E in that case. 
and its the number of sgEth tokens minted as well. 
if interest is earned, sgETH is minted and distributed by sending it to the wsgeth contract, making the virtual price go up. 
this increments curValidatorShares, because sgETH is minted, and curValidatorShares should decrement automatically by that amount when the user withdraws

Reply:
so you mean test cases are invalid?

Reply:
@devlancer412 take a look at the new branch i just created, i fixed them. 
yea test cases were invalid.